tests/parametric/test_dynamic_configuration.py

```diff
@@ -393,6 +366,78 @@ def test_log_injection_enabled(self, library_env, test_agent, test_library):
         assert cfg_state["apply_state"] == 2
 
 
+@rfc("https://docs.google.com/document/d/1SVD0zbbAAXIsobbvvfAEXipEUO99R9RMsosftfe9jx0")
+@scenarios.parametric
+@features.dynamic_configuration
+class TestDynamicConfigV1_ServiceTargets:
+    """Tests covering the Service Target matching of the dynamic configuration feature.
+
+        - ignore mismatching targets
+        - matching service target case-insensitively
+    """
+
+    @parametrize(
+        "library_env",
+        [
+            {
+                **DEFAULT_ENVVARS,
+                # Override service and env
+                "DD_SERVICE": s,
+                "DD_ENV": e,
+            }
+            for (s, e) in [
+                (DEFAULT_ENVVARS["DD_SERVICE"] + "-override", DEFAULT_ENVVARS["DD_ENV"]),
+                (DEFAULT_ENVVARS["DD_SERVICE"], DEFAULT_ENVVARS["DD_ENV"] + "-override"),
+                (DEFAULT_ENVVARS["DD_SERVICE"] + "-override", DEFAULT_ENVVARS["DD_ENV"] + "-override"),
+            ]
+        ],
+    )
+    def test_not_match_service_target(self, library_env, test_agent, test_library):
+        """Test that the library reports an erroneous apply_state when the service targeting is not correct.
+
+        This can occur if the library requests Remote Configuration with an initial service + env pair and then
+        one or both of the values changes.
+
+        We simulate this condition by setting DD_SERVICE and DD_ENV to values that differ from the service
+        target in the RC record.
+        """
+        _set_rc(test_agent, _default_config(TEST_SERVICE, TEST_ENV))
+        cfg_state = test_agent.wait_for_rc_apply_state("APM_TRACING", state=3)
+        assert cfg_state["apply_state"] == 3
+        assert cfg_state["apply_error"] != ""
+
+    @missing_feature(context.library in ["golang"], reason="Not implemented yet")
```

Comment:
```suggestion
```

I think we can rely on the manifest for skipping golang

Reply:
Not really! Go tracer does a case-sensitive check, so I added this annotation to exclude only that case and not the entire class.

Reply:
oh I see! Yep that makes sense then. Maybe we should update the `reason` to include that context?